core/src/test/java/org/elasticsearch/search/sort/FieldSortBuilderTests.java

```diff
@@ -0,0 +1,43 @@
+package org.elasticsearch.search.sort;
+
+import static org.hamcrest.Matchers.stringContainsInOrder;
+
+import java.util.Arrays;
+
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.test.ESTestCase;
+
+public class FieldSortBuilderTests extends ESTestCase {
+
+    private static final String FIELDNAME = "myfield";
+    private static final String QUOTES = "\"";
+
+    public void testToXContent() throws Exception {
+        FieldSortBuilder fieldSortBuilder = prepareFSB();
+        XContentBuilder xc = XContentFactory.contentBuilder(XContentType.JSON);
+        xc.startObject();
+        fieldSortBuilder.toXContent(xc, FieldSortBuilder.EMPTY_PARAMS);
+        xc.endObject();
+
+        assertThat(xc.string(), stringContainsInOrder(
+                Arrays.asList(QUOTES + FIELDNAME + QUOTES, QUOTES + "order" + QUOTES, QUOTES + SortOrder.ASC.toString() + QUOTES)));
+    }
+
+    public void testToString() throws Exception {
+        FieldSortBuilder fsb = prepareFSB();
+
+        String fsbPrint = fsb.toString();
+
+        assertThat(fsbPrint, stringContainsInOrder(
+                Arrays.asList(QUOTES + FIELDNAME + QUOTES, QUOTES + "order" + QUOTES, QUOTES + SortOrder.ASC.toString() + QUOTES)));
```

Comment:
I'd prefer to have a simple `assertEquals` and to a String comparison here. No need for all the constants either, makes it harder to read IMHO.

Reply:
ToString method does a pretty print of the Json content. I didn't want to be bothered about whitespaces or tabs. At least will you concede me an equalToIgnoringWhiteSpace? I think that would do the test less fragile. Something like ...

```java
assertThat(fsbPrint, equalToIgnoringWhiteSpace("{\"myfield\":{\"order\":\"asc\"}}")
```

Or a simple checking like
```java
 assertTrue(fsbPrint.contains("myfield")). 
```
What do you think?

Reply:
Good point, I forgot about the pretty printing. `equalToIgnoringWhiteSpace` sounds good.